### DISCUSSION TITLE
Are there any adverse side-effects that arise when manually altering Container.children

### DISCUSSION THREAD

**Discussion by plemnoooooo** [pixijs/pixijs#9594] [Q&A] [answered]
When altering or setting ```Container.children``` or any child through the array, do any side-effects happen that could cause things to go wrong?

If there is, is there a way to prevent such things from happening?

Thanks!

**Comment by bigtimebuddy** [comment, marked as answer]
If you are manipulating the children's Array object directly (e.g., remove or insert children), it's probably a good idea to take look at what `addChild` and `removeChild` is doing:

https://github.com/pixijs/pixijs/blob/be75cfb6125f572b09ed01af81fca033527646d8/packages/display/src/Container.ts#L157-L176

Specifically it is good to:
* Make sure the parent is assigned if adding a new child
* Invalidate the `_parentID` on the child's transform
* Increment the `_boundsID`
* Optionally emit any events/callbacks

In general, I wouldn't advise this. It's better to use the methods built into Container for managing the `children` array and to treat that array as read-only.

**Reply by plemnoooooo** [reply]
Thanks for telling me how it works!
